src/include/mpir_request.h

```diff
@@ -440,6 +440,9 @@ static inline MPIR_Request *MPIR_Request_create_from_pool(MPIR_Request_kind_t ki
     } else {
         /* FIXME: This fails to fail if debugging is turned off */
         MPL_DBG_MSG(MPIR_DBG_REQUEST, TYPICAL, "unable to allocate a request");
+        /* This assertion will always be false here. The alternative to this is to put checks after
+         * every request creation call. */
+        MPIR_Assert(req != NULL);
     }
```

Comment:
Not sure I agree. First of all, the entire `else` branch can/should be removed and replaced with the `MPIR_Assert`. Second, arguably the *alternative* is a better choice since we can and should return a proper error to the user.

Reply:
I agree that the second option would be better, but it's not reasonable to expect that to be implemented (at least I don't have time to do it). This seemed like the next best thing.

If we want to keep this code, I don't think you can remove then whole `else` block. The code above doesn't return (though I guess it could, but that wouldn't make it more readable to me).

Reply:
What I meant it (and I think you misunderstood):
```
383    req = MPIR_Handle_obj_alloc_unsafe(...);
+        MPIR_Assert(req != NULL);
```
and remove all the `if(req)` branch below.

For proper error handling (later), we could replace the assertion with `if (req == NULL) goto fn_exit;`

Reply:
👍 I see what you mean now. I've updated the PR to reflect this. I'd recommend reviewing with whitespace changes turned off. 😄